src/useradd.c

```diff
@@ -589,6 +589,17 @@ set_defaults(void)
 		goto err_free_def;
 	}
 
+	/*
+	 * Change the permissions of the new file to 644
+	 */
+	if (chmod(new_file, 0644) != 0) {
+		fprintf (stderr,
+		         _("%s: failed to change permissions for %s\n"),
+		         Prog,
+		         new_file);
+		goto err_free_def;
+	}
```

Comment:
This leaks the file descriptor if it fails.  I'll implement an alternative that doesn't leak.  It'll be similar to when I added fmkstemp().

Reply:
Thank you for your review. I'm not an expert on it, I will close this pull request.